client/src/lib/components/TagSelector/TagSelector.svelte

```diff
@@ -0,0 +1,58 @@
+<script lang="ts">
+  import type { SelectOption } from "src/definitions/form";
+  import type { Tag as TagType } from "src/definitions/tag";
+  import { transformTagToSelectOption } from "src/lib/transformers/form";
+  import { createEventDispatcher } from "svelte";
+  import Search from "../Search/Search.svelte";
+  import Tag from "../Tag/Tag.svelte";
+
+  export let tags: TagType[];
+  export let name: string;
+  export let id: string = name;
+
+  export let selectedTags: TagType[] = [];
+
+  const dispatch = createEventDispatcher();
+
+  const handleSearch = (e: CustomEvent<SelectOption>) => {
+    const tag = tags.find((item) => item.id.toString() === e.detail.value);
+
+    const tagHasBeenAlreadySelected = selectedTags.find(
+      (item) => item.id.toString() === e.detail.value
+    );
+
+    if (!tag || tagHasBeenAlreadySelected) return;
+
+    selectedTags = [...selectedTags, tag];
+    dispatch("change", selectedTags);
+  };
+
+  const handleSelectTag = (e: CustomEvent<TagType>) => {
+    selectedTags = selectedTags.filter((item) => item.id !== e.detail.id);
+    dispatch("change", selectedTags);
+  };
+</script>
+
+<div>
+  <Search
+    on:search={handleSearch}
+    {id}
+    {name}
+    options={tags.map(transformTagToSelectOption)}
```

Comment:
Je suis pas convaincu qu'on ait vraiment besoin d'un utilitaire séparé avec des tests unitaires. C'est un niveau d'_indirection_ en plus alors qu'on pourrait se contenter de TS (le paramètre `options` est typé comme `Array<SelectOption>`):

```suggestion
    options={tags.map(tag => ({ label: tag.name, value: tag.id }))}
```

Reply:
De base j'aime bien sortir un maximum de logique des composants pour que cette logique soit déplacée dans des fonctions séparée réutilisable et facilement testable unitairement.

Je comprends ton point mais si tu n'as pas d'objection formelle je préférerais garder cette forme.